r/tools/nixlibs.R

```diff
@@ -96,53 +96,76 @@ try_download <- function(from_url, to_file, hush = quietly) {
-      )
-    libfile <- NULL
-  }
+try_checksum <- function(binary_url, libfile, hush = quietly) {
   # Explicitly setting the env var to "false" will skip checksum validation
   # e.g. in case the included checksums are stale.
   skip_checksum <- env_is("ARROW_R_ENFORCE_CHECKSUM", "false")
   enforce_checksum <- env_is("ARROW_R_ENFORCE_CHECKSUM", "true")
   checksum_path <- Sys.getenv("ARROW_R_CHECKSUM_PATH", "tools/checksums")
   # validate binary checksum for CRAN release only
-  if (!skip_checksum && dir.exists(checksum_path) && is_release ||
-    enforce_checksum) {
-    checksum_file <- sub(".+/bin/(.+\\.zip)", "\\1\\.sha512", binary_url)
-    checksum_file <- file.path(checksum_path, checksum_file)
-    checksum_cmd <- "shasum"
-    checksum_args <- c("--status", "-a", "512", "-c", checksum_file)
-
-    # shasum is not available on all linux versions
-    status_shasum <- try(
-      suppressWarnings(
-        system2("shasum", args = c("--help"), stdout = FALSE, stderr = FALSE)
-      ),
-      silent = TRUE
-    )
+  # we do this in a try so that if it fails for any reason, we don't pollute the log
+  # but then we will return that the checksum failed
+  status <- try(
+    {
+      if (!skip_checksum && dir.exists(checksum_path) && is_release ||
+        enforce_checksum) {
+        checksum_file <- sub(".+/bin/(.+\\.zip)", "\\1\\.sha512", binary_url)
+        checksum_file <- file.path(checksum_path, checksum_file)
+        checksum_cmd <- "shasum"
+        checksum_args <- c("--status", "-a", "512", "-c", checksum_file)
+
+        # shasum is not available on all linux versions, so check help if it exists
+        status_shasum <- try(
+          suppressWarnings(
+            system2("shasum", args = c("--help"), stdout = FALSE, stderr = FALSE)
+          ),
+          silent = TRUE
+        )
 
-    if (inherits(status_shasum, "try-error") || is.integer(status_shasum) && status_shasum != 0) {
-      checksum_cmd <- "sha512sum"
-      checksum_args <- c("--status", "-c", checksum_file)
-    }
+        # if shasum doens't exist, then change the command to sha512sum
+        if (inherits(status_shasum, "try-error") || is.integer(status_shasum) && status_shasum != 0) {
+          checksum_cmd <- "sha512sum"
+          checksum_args <- c("--status", "-c", checksum_file)
+        }
 
-    checksum_ok <- system2(checksum_cmd, args = checksum_args)
+        checksum_ok <- system2(checksum_cmd, args = checksum_args)
 
-    if (checksum_ok != 0) {
-      lg("Checksum validation failed for libarrow: %s/%s", lib, libfile)
-      unlink(libfile)
-      libfile <- NULL
-    } else {
-      lg("Checksum validated successfully for libarrow: %s/%s", lib, libfile)
+        if (checksum_ok != 0) {
+          lg("Checksum validation failed for libarrow: %s/%s", lib, libfile)
+          unlink(libfile)
+          stop("The checksum failed")
+        } else {
+          lg("Checksum validated successfully for libarrow: %s/%s", lib, libfile)
```

Comment:
`lib` isn't available https://github.com/ursacomputing/crossbow/actions/runs/7508393664/job/20443922655#step:13:1612
probably easiest just to pass it as another arg

Reply:
Cleaner would be to drop it from the message here, the next message printed (in `download_binary()`) will include `lib` so it's not needed.

Reply:
In fact, you can probably just say "Checksum validated/failed". (Might be nice to print the checksum here, or even show the output of shasum, in case it's useful to show receipts.)

Reply:
This is done now, thanks